SSH tunnel tests flaky on ci

## Summary

```
 2023-05-27T14:11:17.873744Z ERROR                 main ThreadId(01) testing::slt::cli: crates/testing/src/slt/cli.rs:244: Error while running test `sqllogictests_mysql/tunnels/ssh` error=test fail: statement failed: db error: ERROR: External database validation failed: Connect(Custom { kind: ConnectionAborted, error: "kex_exchange_identification: Connection closed by remote host\r\nConnection closed by ::1 port 46105" })
[SQL] CREATE EXTERNAL DATABASE test_db
	FROM mysql
	TUNNEL test_ssh_tunnel_0
	OPTIONS (
		connection_string = 'mysql://root@172.17.0.3:3306/glaredb_test',
	);
at /home/runner/work/glaredb/glaredb/testdata/sqllogictests_mysql/tunnels/ssh.slt:3
```

I've seen it with the postgres tests too.

### Specifications

## Goals

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -228,8 +228,13 @@ jobs:
 
           echo "-------------------------------- WITHOUT TUNNEL TEST --------------------------------"
           just sql-logic-tests -v --exclude '*/tunnels/ssh'
+
           echo "-------------------------------- WITH TUNNEL TEST --------------------------------"
-          just sql-logic-tests -v '*/tunnels/ssh'
+          # SSH tests are prone to fail if we make a lot of connections at the
+          # same time. Hence, it makes sense to run all the SSH tests one-by-one
+          # in order to test the SSH tunnels (which is our aim).
+          just sql-logic-tests -v --jobs=1 '*/tunnels/ssh'
+
           echo "-------------------------------- RPC TESTS --------------------------------"
           just sql-logic-tests -v --rpc-test \
                                              'sqllogictests/cast/*' \

--- a/crates/testing/tests/sqllogictests/hooks.rs
+++ b/crates/testing/tests/sqllogictests/hooks.rs
@@ -172,6 +172,9 @@ SELECT public_key
         let deadline = Instant::now() + Duration::from_secs(120);
         while Instant::now() < deadline {
             if check_container(container_id).await.is_ok() {
+                // Yes the check completed, but let's just wait a second for the
+                // server to start (if there is anything pending).
+                tokio_sleep(Duration::from_secs(1)).await;
                 return Ok(());
             }
             tokio_sleep(Duration::from_millis(250)).await;

